lib/error_catalog.h

```diff
@@ -97,10 +102,24 @@ class ErrorCatalog {
 
     /// add to the catalog
     /// returns false if the code already exists and forceReplace was not set to true
+    /// @param type      - error/warning/info message
     /// @param errorCode - integer value for the error/warning
     /// @param name      - name for the error. Used to enable/disable all errors of that type
     /// @param forceReplace - override an existing error type in the catalog
-    bool add(int errorCode, const char *name, bool forceReplace = false) {
+    bool add(MessageType type, int errorCode, const char *name, bool forceReplace = false) {
```

Comment:
It would require actually calling `add` in compile-time context. However map::emplace is not constexpr (not even in C++23). Or it would need to be a template, somewhat like this:

```cpp
template<MessageType type, int errCode>
bool add(const char *name, bool forceReplace = flase) {
   static_assert(type !=  MessageType::Error ||
            (errorCode >= ErrorType::ERR_MIN_BACKEND && errorCode < ErrorType::ERR_MAX),
            "Error code out of range");
   ...
}
```

That would work too (I believe). It would be called like this:

```cpp
errorCatalog.add<MessageType::Error, Err::ERR_FOO>("foo");
```

Which is not super-nice, but I'd say it is worth the added safety.

For this to work, the constants would need to be declared in the `.h` file. Probably as

```cpp
static constexpr int ERR_MIN_BACKEND = ...;
```

Reply:
I'll let you decide whether you consider it worth implementing. I think it'd be nice, since runtime `BUG`s are a persistence source of annoyance.

Reply:
Thanks for your suggestions, they really make sense!